supertokens/userIdMapping.go

```diff
@@ -0,0 +1,217 @@
+		ExternalUserId     string
+		ExternalUserIdInfo *string
+	}
+	UnknownMappingError *struct{}
+}
+
+func GetUserIdMapping(userId string, userIdType *string) (GetUserIdMappingResult, error) {
+	if userIdType == nil {
+		userIdType = &userIdTypeDefault
+	}
+	err := validateUserIdType(*userIdType)
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+
+	querier, err := GetNewQuerierInstanceOrThrowError("")
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	cdiVersion, err := querier.GetQuerierAPIVersion()
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	if maxVersion(cdiVersion, "2.15") != cdiVersion {
+		return GetUserIdMappingResult{}, errors.New("Please upgrade the SuperTokens core to >= 3.15.0")
+	}
+
+	resp, err := querier.SendGetRequest("/recipe/userid/map", map[string]string{
+		"userId":     userId,
+		"userIdType": *userIdType,
+	})
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	if resp["status"] == "OK" {
+		var externalUserIdInfo *string = nil
+		if v, ok := resp["externalUserIdInfo"].(string); ok {
+			externalUserIdInfo = &v
+		}
+		return GetUserIdMappingResult{
+			OK: &struct {
+				SupertokensUserId  string
+				ExternalUserId     string
+				ExternalUserIdInfo *string
+			}{
+				SupertokensUserId:  resp["superTokensUserId"].(string),
+				ExternalUserId:     resp["externalUserId"].(string),
+				ExternalUserIdInfo: externalUserIdInfo,
+			},
+		}, nil
+	} else {
+		return GetUserIdMappingResult{
+			UnknownMappingError: &struct{}{},
+		}, nil
+	}
+}
+
+type DeleteUserIdMappingResult struct {
+	OK *struct {
+		DidMappingExist bool
+	}
+}
+
+func DeleteUserIdMapping(userId string, userIdType *string, force bool) (DeleteUserIdMappingResult, error) {
+	if userIdType == nil {
+		userIdType = &userIdTypeDefault
+	}
+	err := validateUserIdType(*userIdType)
+	if err != nil {
+		return DeleteUserIdMappingResult{}, err
+	}
+
+	querier, err := GetNewQuerierInstanceOrThrowError("")
+	if err != nil {
+		return DeleteUserIdMappingResult{}, err
+	}
+	cdiVersion, err := querier.GetQuerierAPIVersion()
+	if err != nil {
+		return DeleteUserIdMappingResult{}, err
+	}
+	if maxVersion(cdiVersion, "2.15") != cdiVersion {
+		return DeleteUserIdMappingResult{}, errors.New("Please upgrade the SuperTokens core to >= 3.15.0")
+	}
+
+	resp, err := querier.SendPostRequest("/recipe/userid/map/remove", map[string]interface{}{
+		"userId":     userId,
+		"userIdType": userIdType,
+		"force":      force,
+	})
+	if err != nil {
+		return DeleteUserIdMappingResult{}, err
+	}
+	return DeleteUserIdMappingResult{
+		OK: &struct{ DidMappingExist bool }{
+			DidMappingExist: resp["didMappingExist"].(bool),
+		},
+	}, nil
+}
+
+type UpdateOrDeleteUserIdMappingInfoResult struct {
+	OK                  *struct{}
+	UnknownMappingError *struct{}
+}
+
+func UpdateOrDeleteUserIdMappingInfo(userId string, userIdType *string, externalUserIdInfo *string) (UpdateOrDeleteUserIdMappingInfoResult, error) {
```

Comment:
need to add a test where you delete the externalUserIdInfo and then call getUserIdMapping and see what happens

Reply:
Done